evaluation/model_selector.py

```diff
@@ -0,0 +1,139 @@
+import pandas as pd
+import math
+import itertools
+from .plotutils.radar import make_radar_plot
+from typing import List, Tuple, Generator
+from prettytable import PrettyTable
+
+# TODO: figure out how to horizontally import model and data
+# for type annotations
+class EvaluationTable(dict):
+    def __init__(self, *args, **kw):
+        super(EvaluationTable, self).__init__(*args, **kw)
+    def __str__(self):
+        out = PrettyTable()
+        metrics = list(self[list(self.keys())[0]].keys())
+        out.field_names = ['Model'] + metrics
+        for model_name in self:
+            to_add = [model_name] + [self[model_name][metric] for metric in metrics]
+            out.add_row(to_add)
+        out.float_format = f'.3'
+        return out.__str__()
+    def __repr__(self):
+        return self.__str__()
+
+
+
+
+def model_selector(models: List,
```

Comment:
This along with some other methods below should be in one class really

Reply:
Then the `smart_model_selector` should probably be a child class of `model_selector`

Reply:
I instead created a `ModelSelector` class which has `run()`, `run_halving()`, and `visualize()` methods.